src/Kiota.Builder/Plugins/PluginsGenerationService.cs

```diff
@@ -78,6 +78,13 @@ public async Task GenerateManifestAsync(CancellationToken cancellationToken = de
         {
             var manifestFileName = $"{Configuration.ClientClassName.ToLowerInvariant()}-{pluginType.ToString().ToLowerInvariant()}";
             var manifestOutputPath = Path.Combine(Configuration.OutputPath, $"{manifestFileName}{ManifestFileNameSuffix}");
+
+            // Check directly the output when workspace is not used
+            if (Configuration.Operation == ConsumerOperation.Add && Configuration.NoWorkspace && File.Exists(manifestOutputPath))
```

Comment:
I believe we're mixing concerns with clean output here. I'd suggest removing this block

Reply:
Done